### PR TITLE
add crypto suite and UT

### PR DESCRIPTION
**Pull request by auxten**
add crypto suite port from btcec and UT

### CODE REVIEW

File: crypto/cipher.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	length := len(src)
+	padLength := int(src[length-1])
+	if padLength > aes.BlockSize || length < aes.BlockSize {
+		return nil, errInvalidPadding
+	}
+
+	return src[:length-padLength], nil
+}
+
+// EncryptAndSign(inputPublicKey, inData) MAIN PROCEDURE:
```
Comment (covenantdb) [inline crypto/cipher.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
comment on exported function EncryptAndSign should be of the form "EncryptAndSign ..."

File: crypto/cipher.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+//	1. newPrivateKey, newPubKey := genSecp256k1Keypair()
+//	2. encKey, HMACKey := SHA512(ECDH(newPrivateKey, inputPublicKey))
+//	3. PaddedIn := PKCSPadding(in)
+//	4. OutBytes := IV + newPubKey + AES-256-CBC(encKey, PaddedIn) + HMAC-SHA-256(HMACKey)
+func EncryptAndSign(inputPublicKey *btcec.PublicKey, inData []byte) ([]byte, error) {
+	return btcec.Encrypt(inputPublicKey, inData)
+}
+
+// DecryptAndCheck(inputPrivateKey, inData) MAIN PROCEDURE:
```
Comment (covenantdb) [inline crypto/cipher.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
comment on exported function DecryptAndCheck should be of the form "DecryptAndCheck ..."

File: crypto/asymmetric/doc.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+This package provides the data structures and functions implementing the
+crypto/elliptic Curve interface in order to permit using these curves
+with the standard crypto/ecdsa package provided with go. Helper
+functionality is provided to parse signatures and public keys from
+standard formats.  It was designed for use with btcd, but should be
+general enough for other uses of elliptic curve crypto.  It was originally based
+on some initial work by ThePiachu, but has significantly diverged since then.
+*/
+
```
Comment (covenantdb) [inline crypto/asymmetric/doc.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
package comment is detached; there should be no blank lines between it and the package statement

File: crypto/asymmetric/keypair.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package asymmetric
+
+import (
+	"github.com/btcsuite/btcd/btcec"
+	log "github.com/sirupsen/logrus"
+)
+
+func GenSecp256k1Keypair() (privateKey *btcec.PrivateKey, publicKey *btcec.PublicKey, err error) {
```
Comment (covenantdb) [inline crypto/asymmetric/keypair.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function GenSecp256k1Keypair should have comment or be unexported

File: crypto/hash/hashfuncs.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+// Copyright (c) 2016-2017 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package hash
+
+import "crypto/sha256"
+
+// HashB calculates hash(b) and returns the resulting bytes.
+func HashB(b []byte) []byte {
```
Comment (covenantdb) [inline crypto/hash/hashfuncs.go:11]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
func name will be used as hash.HashB by other packages, and that stutters; consider calling this B

File: crypto/hash/hashfuncs.go [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+	hash := sha256.Sum256(b)
+	return hash[:]
+}
+
+// HashH calculates hash(b) and returns the resulting bytes as a Hash.
+func HashH(b []byte) Hash {
```
Comment (covenantdb) [inline crypto/hash/hashfuncs.go:17]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
func name will be used as hash.HashH by other packages, and that stutters; consider calling this H

**Review by zeqing-guo — COMMENTED** [review]
greate!

**Review by zeqing-guo — COMMENTED** [review]
greate

**Review by zeqing-guo — APPROVED** [review]

**Pull request merged**